### PR TITLE
[ACS-8215] Filter out dependencies of dependencies in licenses.ts script

### PR DESCRIPTION
**Pull request by tomson7777**
JIRA ticket: https://hyland.atlassian.net/browse/ACS-8215

Script omits child dependencies of any top-level library.

### CODE REVIEW

File: lib/cli/scripts/licenses.ts [hunk 1]
```diff
@@ -148,9 +152,19 @@ export default function main(_args: string[], workingDir: string) {
                     console.error(err);
                     reject(err);
                 } else {
+                    const filteredPackages = Object.keys(directDependencies).reduce((obj, packageName) => {
+                        obj[packageName] = packages[packageName];
+                        return obj;
+                    }, {});
+
+                    // eslint-disable-next-line no-console
+                    console.log({ filteredPackages });
```
Comment (AleksanderSklorz) [inline lib/cli/scripts/licenses.ts:161]: Do we need this console log and that one below?

File: docs/license-info/license-info-7.0.0-alpha.2.md
Comment (AleksanderSklorz) [inline docs/license-info/license-info-7.0.0-alpha.2.md:1]: was it intentional to delete this file?

File: lib/cli/scripts/licenses.ts (hunk 1 above)
Comment (DaryaBalvanovich) [reply lib/cli/scripts/licenses.ts]: removed

File: docs/license-info/license-info-7.0.0-alpha.2.md
Comment (DaryaBalvanovich) [reply docs/license-info/license-info-7.0.0-alpha.2.md:1]: reverted

File: lib/cli/scripts/licenses.ts [hunk 2]
```diff
@@ -148,29 +151,34 @@ export default function main(_args: string[], workingDir: string) {
...
-                            }
+
+                        const lastAtSignPos = packageName.lastIndexOf('@');
+                        const basePackageName = packageName.substring(0, lastAtSignPos);
+
+                        if (mainDependencies.includes(basePackageName)) {
+                            pack['licenseExp'] = pack['licenses']
+                                .toString()
+                                .replace(/\*/g, '')
+                                .replace(/[a-zA-Z0-9\-.]+/g, (match) => {
```
Comment (AleksanderSklorz) [inline lib/cli/scripts/licenses.ts:167]: seems like type of match is any, could you please set it to string as it seems to be string?
Screenshot 2024-10-16 at 12 04 46

Comment (DaryaBalvanovich) [reply lib/cli/scripts/licenses.ts]: added

**Review by AleksanderSklorz — APPROVED** [review]

File: lib/cli/scripts/licenses.ts [hunk 3]
```diff
@@ -148,29 +151,34 @@ export default function main(_args: string[], workingDir: string) {
...
-
-                        if (!pack['repository']) {
-                            const lastAtSignPos = packageName.lastIndexOf('@');
-                            const mainName = packageName.substring(0, lastAtSignPos);
-
-                            if (missingRepositories[mainName]) {
-                                pack['repository'] = missingRepositories[mainName];
-                            }
+
+                        const lastAtSignPos = packageName.lastIndexOf('@');
```
Comment (MichalKinas) [inline lib/cli/scripts/licenses.ts:160]: Did you had a chance to fix those changes? And if so how did you test it?

Comment (DaryaBalvanovich) [reply lib/cli/scripts/licenses.ts:160]: Unfortunately I can't support this ticket anymore that's why I asked @MateuszHY to double check with someone who can take care and retest it. Also you can talk with @eromano about running scripts

I changed `node_modules/@alfresco/adf-cli/scripts/licenses.js` then run `scripts/release.sh` and see output.
After I checked that it works correctly (creates new file with 38 dependencies lines) I applied these changes to `lib/cli/scripts/licenses.ts`.
Also you can check with `node dist/libs/cli/scripts/licenses.js` command

Comment (MichalKinas) [reply lib/cli/scripts/licenses.ts:159]: The functionality was not working as is, @AleksanderSklorz did a great job investigating the issue and we added required fix.

**Pull request merged**